tests/utilities/test_fetching.py

```diff
@@ -84,3 +95,148 @@ def test_misconfiguration_error():
 
     iter(fetcher)
     assert fetcher.loader_iters
+
+
+def get_cycles_per_ms() -> float:
+    """
+    Measure and return approximate number of cycles per millisecond for torch.cuda._sleep
+    Copied from: github.com/pytorch/pytorch/blob/master/test/test_cuda.py
+    """
+
+    def measure() -> float:
+        start = torch.cuda.Event(enable_timing=True)
+        end = torch.cuda.Event(enable_timing=True)
+        start.record()
+        torch.cuda._sleep(1000000)
+        end.record()
+        end.synchronize()
+        cycles_per_ms = 1000000 / start.elapsed_time(end)
+        return cycles_per_ms
+
+    # Get 10 values and remove the 2 max and 2 min and return the avg.
+    # This is to avoid system disturbance that skew the results, e.g.
+    # the very first cuda call likely does a bunch of init, which takes
+    # much longer than subsequent calls.
+    #
+    # Tested on both Tesla V100, Quadro GP100, Titan RTX, RTX 3090 GPUs
+    # and seems to return stable values. Therefore, we enable caching
+    # using lru_cache decorator above.
+    num = 10
+    vals = []
+    for _ in range(num):
+        vals.append(measure())
+    vals = sorted(vals)
+    stats = vals[2 : num - 2]
+    return sum(stats) / len(stats)
+
+
+BATCH_SIZE = 128
+EMB_SZ = 100
+EMB_DIM = 64
+
+
+class RandomIndicesDataset(Dataset):
+    def __getitem__(self, index):
+        return torch.randint(EMB_DIM, [BATCH_SIZE])
+
+    def __len__(self):
+        return 16
+
+
+class RecommenderModel(BoringModel):
+    def __init__(self):
+        super().__init__()
+        self.layer = None
+        self.local_embedding = torch.nn.Embedding(EMB_SZ, EMB_DIM)
+        self.CYCLES_PER_MS = int(get_cycles_per_ms())
+
+    def forward(self, indices: torch.Tensor):
+        result = self.local_embedding(indices)
+        return result
+
+    def on_after_batch_transfer(self, batch: Any, dataloader_idx: int) -> Any:
+        # emulate heavy routine
+        torch.cuda._sleep(self.CYCLES_PER_MS * 100)
+        return batch
+
+    def training_step_end(self, training_step_outputs):
+        # emulate heavy routine
+        torch.cuda._sleep(self.CYCLES_PER_MS * 100)
+        return training_step_outputs
+
+    def configure_optimizers(self):
+        return torch.optim.SGD(self.parameters(), lr=0.1)
+
+    def train_dataloader(self):
+        return DataLoader(RandomIndicesDataset(), batch_size=4)
+
+    def val_dataloader(self):
+        return DataLoader(RandomIndicesDataset(), batch_size=4)
+
+    def test_dataloader(self):
+        return DataLoader(RandomIndicesDataset(), batch_size=4)
+
+
+@RunIf(min_gpus=1, min_torch="1.8.0")
+def test_trainer_num_prefetch_batches(tmpdir):
```

Comment:
@tchaton how expensive are these tests? I see we are waiting to simulate a heavy cuda call?

Reply:
This test takes 7 sec. I optimized it as much as possible,

Reply:
I strongly recommend to reconsider this. 7 seconds is an eternity considering we have 1000's of tests. 
If we can't unit test it < 1s, should we only run it locally perhaps?